### ISSUE TITLE
Postcss import variables

### ISSUE THREAD

**Issue by fgarcia** [MoOx/phenomic#846]
I've been using `@import 'my.css'` for a while now with Phenomic, but I've just realized that everything is OK except for variables, they are never imported

I tried adding `postcss-import` plugin to the webpack config file, but that causes extra errors, probably a conflict with `postcss-next`

Probably this is not exclusive to Phenomic, but since this shows up in the default template I thought there should be an issue about it.

**Comment by DavidWells** [comment]
Yeah make sure the order of your postcss transforms are correct. 

I personally use postcss-simple-vars for variables and load it before the postcss-import plugin

**Comment by khrome83** [comment]
I got this working with no issues. This is modification to the base theme that I started with.

1. Install `postcss-import` through NPM
```
  npm install postcss-import --save-dev
```

2. Update this line to pass in `webp`

```
  const postcssPlugins = (webp) => [
```

3. Add `postcss-import` before `postcss-cssnext`. Make sure to use `webp` that we passed in earlier.
```
    require("stylelint")(),
    require("postcss-import")({ addDependencyTo: webp }),
    require("postcss-cssnext")({
```

4. In any css file, add `@import "../../css/_variables.css";` being a path to your other CSS file.

__Note__

If your file only contains variables like the example below, you will need to have some common normal selector so you don't get a `empty file` error.

```css
:root {
  --background-color: #fff;
}
```

Fixed this by adding this to the end of the file.
```css
html {
  color: inherit;
}
```

**Comment by MoOx** [comment]
You should not have to add dumb content. postcss-import should inline raw files, so not transformed by cssnext ( :root are in the file when postcss-import is doing its job).

**Comment by khrome83** [comment]
Agree. Was giving me empty file error though.

Get Outlook for iOS




On Wed, Nov 9, 2016 at 9:57 AM -0600, "Maxime Thirouin" <notifications@github.com> wrote:












You should not have to add dumb content. postcss-import should inline raw files, so not transformed by cssnext ( :root are in the file when postcss-import is doing its job).

**Comment by daaain** [comment]
Thanks for the writeup @khrome83, your method worked for me in the latest version with the small modification of having to add the webpack instance in `webpack.config.js` like this:

```js
return require(postcssPluginFile)(config, webpackInstance)
```

...and then adding to `postcss.config.js` like this:

```js
  module.exports = (config, webpackInstance) => [
    require("stylelint")(),
    require("postcss-import")({ addDependencyTo: webpackInstance }),
```

Everything seems to be working, but I do get odd messages like:

```
⠸ Updating
src/index.global.css
undefined [undefined]
```

**Comment by MoOx** [comment]
With latest version of postcss-import, addDependencyTo is useless and automatically handled, no need to pass webpack instance.

**Comment by daaain** [comment]
Ah cool, it does work without, thanks @MoOx!

Still getting these `undefined [undefined]` messages though, but while things work I'm not too bothered :)

**Comment by MoOx** [comment]
Where do you get this message? In terminal? Can you try to disable stylelint for example so we can find the culprit? Maybe it's related to the new postcss message api poke @ai @RyanZim

**Comment by daaain** [comment]
Yes, on the terminal.

Same thing without stylelint, but it does go away if I remove postcss-reporter.

**Comment by MoOx** [comment]
yeah so "something" is adding weird messages (or postcss-reporter is displaying something it should not)

**Comment by khrome83** [comment]
Yeah I can confirm that I have been seeing this as well. 

Get Outlook for iOS




On Thu, Dec 1, 2016 at 5:32 AM -0600, "Maxime Thirouin" <notifications@github.com> wrote:












yeah so "something" is adding weird messages (or postcss-reporter is displaying something it should not)

**Comment by RyanZim** [comment]
FYI, I'm swamped with other things ATM, so won't get a chance to look at this in the near future. If you  can make a minimum, verifiable example, I might be able to look at it sometime.

**Comment by bloodyowl** [comment]
Going to be fixed with #925 (moving to user-land)

**Comment by MoOx** [comment]
User land thing.
